iothub/service/src/Exceptions/IotHubServiceException.cs

```diff
@@ -79,5 +78,21 @@ public override void GetObjectData(SerializationInfo info, StreamingContext cont
             info.AddValue(IsTransientValueSerializationStoreName, IsTransient);
             info.AddValue(TrackingIdSerializationStoreName, TrackingId);
         }
+
+        private bool DetermineIfTransient(HttpStatusCode statusCode, IotHubErrorCode errorCode)
+        {
+            switch (errorCode)
```

Comment:
I'd say we should label any 500 level error (ServerError, ServiceUnavailable) as transient, and I'd argue "DeviceNotOnline" is retriable too since the device's state will likely return to online soon if the customer has a retry policy in place

Reply:
> "DeviceNotOnline" is retriable too since the device's state will likely return to online soon if the customer has a retry policy in place

"DeviceNotOnline" can also happen when the direct method callback isn't registered on the device, which seems not able to recover via the custom retry policy. Should we still consider this as retriable?

Reply:
Yeah, probably retriable for DeviceNotOnline.